dask/dataframe/core.py

```diff
@@ -4204,17 +4204,37 @@ def _take_last(a, skipna=True):
         Whether to exclude NaN
 
     """
+
+    def _collect_series_last_valid(s):
+        size = len(s)
```

Comment:
If this is only going to be used once then lets just inline it.

Reply:
Used twice, so will keep as a function consistent with other suggestions in your review.

Reply:
I meant the definition of `size = len(s)`

```suggestion
```